superbench/benchmarks/micro_benchmarks/ib_validation_performance.py

```diff
@@ -0,0 +1,320 @@
+            help='The path of config file on the target machines',
+        )
+        self._parser.add_argument(
+            '--bidirectional', action='store_true', default=False, help='Measure bidirectional bandwidth.'
+        )
+        self._parser.add_argument(
+            '--gpu_index', type=int, default=None, required=False, help='Test Use GPUDirect with the gpu index.'
+        )
+        self._parser.add_argument(
+            '--hostfile',
+            type=str,
+            default='/root/hostfile',
+            required=False,
+            help='The path of hostfile on the target machines',
+        )
+
+    def __one_to_many(self, n):
+        config = []
+        for client in range(n):
+            row = []
+            for server in range(n):
+                if server != client:
+                    pair = '{},{}'.format(server, client)
+                    row.append(pair)
+            row = ';'.join(row)
+            config.append(row)
+        return config
+
+    def __many_to_one(self, n):
+        config = []
+        for server in range(n):
+            row = []
+            for client in range(n):
+                if server != client:
+                    pair = '{},{}'.format(server, client)
+                    row.append(pair)
+            row = ';'.join(row)
+            config.append(row)
+        return config
+
+    def __roundRobin(self, candidates):
+        res = []
+        count = len(candidates)
+        robin = candidates
+        first = [candidates[0]]
+        for _ in range(count - 1):
+            pairs = [[robin[i], robin[count - i - 1]] for i in range(0, (count + 1) // 2)]
+            res.append(pairs)
+            robin = robin[1:]
+            robin = first + robin[1:] + robin[:1]
+        return res
+
+    def __fully_one_to_one(self, n):
+        config = []
+        if n % 2 == 1:
+            return config
+        candidates = list(range(n))
+        res = self.__roundRobin(candidates)
+        for line in res:
+            row = []
+            for pair in line:
+                row.append('{},{}'.format(pair[0], pair[1]))
+            row = ';'.join(row)
+            config.append(row)
+        return config
+
+    def gen_traffic_pattern(self, n, mode, config_file_path):
+        """Generate traffic pattern into config file.
+
+        Args:
+            n (int): the number of nodes.
+            mode (str): the traffic mode, including 'one-to-one', 'one-to-many', 'many-to-one'.
+            config_file_path (str): the path of config file to generate
+        """
+        config = []
+        if mode == 'one-to-many':
+            config = self.__one_to_many(n)
+        elif mode == 'many-to-one':
+            config = self.__many_to_one(n)
+        elif mode == 'one-to-one':
+            config = self.__fully_one_to_one(n)
+        with open(config_file_path, 'w') as f:
+            for line in config:
+                f.write(line + '\n')
+
+    def _preprocess(self):    # noqa: C901
```

Comment:
Is it possible to optimize the code a little bit? There is too many # noqa: C901.